File: daogen/projects/5223f7ae28afd214e875aca5/src/main/java/com/ctrip/platform/international/daogen/dao/PersonDAO.java
package com.ctrip.platform.international.daogen.dao;

import java.sql.ResultSet;
import java.util.Arrays;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import com.ctrip.platform.international.daogen.dao.common.AbstractDAO;
import com.ctrip.platform.international.daogen.dao.exception.ParametersInvalidException;
import com.ctrip.platform.international.daogen.dao.param.Parameter;

public class PersonDAO extends AbstractDAO {
	
	private static final Logger logger = LoggerFactory.getLogger(PersonDAO.class);

	
	//None									
	public  int  delete(Parameter... params)
			throws Exception {

		
		final String sql = "DELETE FROM Person    WHERE  ID = ? AND Address = ? AND Name > ? ";
		

		return super.execute(null, sql, 0, params);
	}
	
	//None									
	public  ResultSet  get(Parameter... params)
			throws Exception {

		
		Arrays.sort(params);

		String[] placeHolder = new String[params.length];

		for (int i = 0; i < params.length; i++) {
			//First set the place holder to just one parameter
			placeHolder[i] = "?";
			
			//if and only if user pass in valid parameter, we
			//format the place holder for him
			if (params[i].getValue().isArrayValue()) {
				int batchSize = params[i].getValue().asArrayValue().size();

				if (batchSize > 0) {
					StringBuilder inClause = new StringBuilder();
					inClause.append('(');
					for (int j = 0; j < batchSize; j++) {
						inClause.append('?');
						if (j != batchSize - 1) {
							inClause.append(',');
						}
					}
					inClause.append(')');
					placeHolder[i] = inClause.toString();
				}
			}
			
		}
		final String sql = String
				.format("SELECT ID,Address,Name,Telephone,Age,Gender,Birth FROM Person  WHERE  Age In %s ",
						(Object[]) placeHolder);
		

		return super.fetch(null, sql, 0, params);
	}
	
	//None									
	public  ResultSet  getAllByIDAddress(Parameter... params)
			throws Exception {

		
		final String sql = "SELECT ID,Address,Name,Telephone,Age,Gender,Birth FROM Person   WHERE  ID = ? AND Address != ? ";
		

		return super.fetch(null, sql, 0, params);
	}
	

	
	//None									
	public  int  insert(Parameter... params)
			throws Exception {
		

		final String spName = "spa_Person_i";
	

		
			return super.executeSp(null, spName, 0, params);
		
	}
	
	//None									
	public  int  set(Parameter... params)
			throws Exception {
		

		final String spName = "spa_Person_u";
	

		
			return super.executeSp(null, spName, 0, params);
		
	}
	

}

File: daogen/projects/5223f7ae28afd214e875aca5/PersonDAO.cs
using System;
using System.Data;
using platform.dao.client;
using platform.dao.param;

namespace platform.international.daogen
{
    public class PersonDAO
    {
        public static IClient client = ClientFactory.CreateDbClient("platform.dao.providers.SqlDatabaseProvider,platform.dao",
            "Server=testdb.dev.sh.ctriptravel.com,28747;Integrated Security=sspi;database=SysDalTest;");

        public static IClient dasClient = ClientFactory.CreateDasClient("SysDalTest", "user=kevin;password=kevin");

        
        // None
        public IDataReader getAllByPk(int iD)
        {
            try
            {
                StatementParameterCollection parameters = new StatementParameterCollection();

                //parameters.Add(new StatementParameter { Index = 1, DbType = DbType.Int64, Value = pk });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@ID", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.Int32, 
                        Value = iD 
                        });
                
                
                string sql = "SELECT ID,Address,Name,Telephone,Age,Gender,Birth FROM Person  WHERE  ID = @ID "   ;

                //return client.Fetch(sql, parameters);
                
                return dasClient.Fetch(sql, parameters);
                
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        
        // None
        public int insert(int iD, string address, string name, string telephone, int age, int gender, DateTime birth)
        {
            try
            {
                StatementParameterCollection parameters = new StatementParameterCollection();

                //parameters.Add(new StatementParameter { Index = 1, DbType = DbType.Int64, Value = pk });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@ID", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.Int32, 
                        Value = iD 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Address", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.String, 
                        Value = address 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Name", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.String, 
                        Value = name 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Telephone", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.String, 
                        Value = telephone 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Age", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.Int32, 
                        Value = age 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Gender", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.Int32, 
                        Value = gender 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Birth", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.DateTime, 
                        Value = birth 
                        });
                
                
                string sql = "INSERT INTO Person (ID,Address,Name,Telephone,Age,Gender,Birth) VALUES (@ID,@Address,@Name,@Telephone,@Age,@Gender,@Birth)"   ;

                //return client.Fetch(sql, parameters);
                
                return dasClient.Execute(sql, parameters);
                
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        

        
        // None
        public int set(int iD, string address, string name, string telephone, int age, int gender, DateTime birth)
        {
            try
            {
                StatementParameterCollection parameters = new StatementParameterCollection();

                //parameters.Add(new StatementParameter { Index = 1, DbType = DbType.Int64, Value = pk });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@ID", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.Int32, 
                        Value = iD 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Address", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.String, 
                        Value = address 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Name", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.String, 
                        Value = name 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Telephone", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.String, 
                        Value = telephone 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Age", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.Int32, 
                        Value = age 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Gender", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.Int32, 
                        Value = gender 
                        });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@Birth", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.DateTime, 
                        Value = birth 
                        });
                
                
                string sp = "spa_Person_u"   ;

                //return client.Execute(sql, parameters);

                return dasClient.ExecuteSp(sp, parameters);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        
        // None
        public int delete(int iD)
        {
            try
            {
                StatementParameterCollection parameters = new StatementParameterCollection();

                //parameters.Add(new StatementParameter { Index = 1, DbType = DbType.Int64, Value = pk });
                
                    parameters.Add(new StatementParameter { 
                        Name = "@ID", 
                        Direction = ParameterDirection.Input, 
                        DbType = DbType.Int32, 
                        Value = iD 
                        });
                
                
                string sp = "spa_Person_d"   ;

                //return client.Execute(sql, parameters);

                return dasClient.ExecuteSp(sp, parameters);
            }
            catch (Exception ex)
            {
                throw ex;
            }
        }
        
    }
}
